vncdotool/api.py

```diff
@@ -137,14 +141,28 @@ def connect(
     http://twistedmatrix.com/documents/13.0.0/core/howto/choosing-reactor.html
     for a better method of intergrating vncdotool.
     """
+
+
+
     if not reactor.running:
+
+        # For api.shutdown to kill reactor threads before trying to exit due to an Exception
+        sys_excepthook = sys.excepthook
+        def ensure_reactor_stopped(etype, value, traceback):
```

Comment:
Nit: Missing type annotation
```suggestion
        def ensure_reactor_stopped(etype: Type[BaseException], value: BaseException, traceback: TracebackType) -> None:
```

That will require some additional `imports` *before* line 12:
```python
from types import TracebackType

```